app/src/test/kotlin/fr/free/nrw/commons/logging/FileLoggingTreeUnitTest.kt

```diff
@@ -0,0 +1,143 @@
+package fr.free.nrw.commons.logging
+
+import android.util.Log
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import org.junit.Before
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.powermock.reflect.Whitebox
+import org.slf4j.Logger
+import java.lang.reflect.Method
+import java.util.concurrent.Executor
+
+class FileLoggingTreeUnitTest {
+
+    private lateinit var fileLoggingTree: FileLoggingTree
+
+    @Mock
+    private lateinit var executor: Executor
+
+    @Mock
+    private lateinit var logger: Logger
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        fileLoggingTree = FileLoggingTree(
+            Log.VERBOSE,
+            "test",
+            "test",
+            1000,
+            executor
+        )
+        Whitebox.setInternalState(fileLoggingTree, "logger", logger)
+    }
+
+    @Test
+    fun testSetLogLevel() {
```

Comment:
How the class stores the log level internally and what it does with it are 2 different things; I dont care too much that calling a setter stores a value, what I do care is that calls to the logger are *filtered* by that log level afterward.  I think this test would be more informative if the verifications demonstrate that changing the log level means the internal logger is called (or not) depending on the value.